routes/books.js

```diff
@@ -1,10 +1,140 @@
 'use strict';
 
 const express = require('express');
+const knex = require('../knex');
+
 
 // eslint-disable-next-line new-cap
 const router = express.Router();
 
 // YOUR CODE HERE
 
+router.get('/books', (_req, res, next) => {
+  knex('books')
+    .orderBy('title', 'ASC')
+    .select('id', 'title', 'author', 'genre', 'description', 'cover_url as coverUrl', 'created_at as createdAt', 'updated_at as updatedAt')
+    .then((books) => {
+      res.send(books);
+    })
+    .catch((err) => {
+      next(err);
+    });
+});
+
+
+router.get('/books/:id', (req, res, next) => {
+  knex('books')
+    .where('id', req.params.id)
+    .first()
+    .select('id', 'title', 'author', 'genre', 'description', 'cover_url as coverUrl', 'created_at as createdAt', 'updated_at as updatedAt')
+    .then((book) => {
+      if (!book) {
+        return next(err);
+      }
+
+      res.send(book);
+    })
+    .catch((err) => {
+      next(err);
+    });
+});
+
+
+router.post('/books', (req, res, next) => {
+  knex('books')
```

Comment:
be sure to make sure that all the properties in the body are there, and are of the right type.